File: src/main/java/com/project/pantau/dto/report/QueueReportResponse.java
package com.project.pantau.dto.report;

import com.project.pantau.dto.category.CategoryResponse;
import com.project.pantau.enums.ReportStatus;
import lombok.Builder;

import java.time.LocalDateTime;
import java.util.UUID;

@Builder
public record QueueReportResponse(
        UUID id,
        CategoryResponse category,
        String description,
        String photoUrl,
        ReportStatus status,
        Double latitude,
        Double longitude,
        Double distanceMeter,
        LocalDateTime createdAt
) {
}

File: backend/internal/dto/report/queue_report_response.go
package report

import (
	"github.com/google/uuid"
	"pantau/internal/dto/category"
	"pantau/internal/enums"
	"time"
)

type QueueReportResponse struct {
	ID            uuid.UUID                 `json:"id"`
	Category      category.CategoryResponse `json:"category"`
	Description   *string                   `json:"description"`
	PhotoURL      *string                   `json:"photo_url"`
	Latitude      *float64                  `json:"latitude"`
	Longitude     *float64                  `json:"longitude"`
	DistanceMeter *float64                  `json:"distance_meter"`
	Status        enums.ReportStatus        `json:"status"`
	CreatedAt     time.Time                 `json:"created_at"`
}
